frontend/src/components/archiveform/ArchiveForm.tsx

```diff
@@ -70,29 +70,34 @@ export function ArchiveForm({
   const [archiveMeta, setArchiveMeta] = useState<ArchiveMeta | null>(null);
 
   /** Form field values. */
-  const [formState, setFormState] = useState<ArchiveFormData>(() => {
-    const resultaatTypeBrondDatumValues =
-      resultaatType.brondatumArchiefprocedure || {};
-    return {
-      resultaattypeomschrijving: resultaatType.resultaattypeomschrijving,
-      selectielijstklasse: resultaatType.selectielijstklasse,
-
-      afleidingswijze: "afgehandeld",
-      datumkenmerk: "",
-      einddatumBekend: false,
-      objecttype: "",
-      registratie: "",
-      procestermijn: "",
-      ...resultaatTypeBrondDatumValues,
-    };
+  const [formState, setFormState] = useState<ArchiveFormData>({
+    resultaattypeomschrijving:
+      resultaatType.resultaattypeomschrijving ||
+      resultaattypeomschrijvingOptions?.[0].value,
+
+    selectielijstklasse:
+      resultaatType.selectielijstklasse ||
+      selectielijstklasseOptions?.[0].value,
+
+    afleidingswijze: resultaatType.brondatumArchiefprocedure?.afleidingswijze,
+    datumkenmerk: "",
+    einddatumBekend: false,
+    objecttype: "",
+    registratie: "",
+    procestermijn: "",
+    ...(resultaatType.brondatumArchiefprocedure ||
+      // @ts-expect-error - FIXME: Backend does not properly apply camelCase.
+      resultaatType.brondatum_archiefprocedure ||
```

Comment:
When does this happen? 🤔 When I look at the response to the GET of a zaaktype, the resultaattypen have the right camelisation?